etcdmain: change default GOMAXPROCS when compiling in go1.5

yichengq opened the pull request:
Change default GOMAXPROCS to the number of CPUs available when compiling
in go1.5

This matches how go 1.5 works:
https://docs.google.com/document/d/1At2Ls5_fhJQ59kDK2DFVhFu3g5mATSXqqV5QrxinasI/edit

This is a behavior change in etcd 2.2, and should be mentioned in release note.

xiang90 commented on etcdmain/gomaxprocs.go:
```diff
@@ -0,0 +1,32 @@
+// Copyright 2015 CoreOS, Inc.
```
can we move this to pkg/runtime?

yichengq commented:
When I am addressing comments, I find another method that could reach the same result that I want with fewer codes, and I think it is better. Instead of setting it according to what different go versions do, it exposes the value set by go.

I will update the commit message if the method is good.

xiang90 commented:
LGTM
